pkg/api/rest/delete.go

```diff
@@ -89,5 +92,13 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx api.Context, obj runtime.Obje
 	now := unversioned.NewTime(unversioned.Now().Add(time.Second * time.Duration(*options.GracePeriodSeconds)))
 	objectMeta.DeletionTimestamp = &now
 	objectMeta.DeletionGracePeriodSeconds = options.GracePeriodSeconds
+	// If it'a first graceful deletion we're going to update the object first,
```

Comment:
There are some misspellings, I also changed last sentence. How about this: "If it's the first graceful deletion we are going to set the DeletionTimestamp to non-nil. Controllers of the object that's being deleted shouldn't take any nontrivial actions, hence its behavior changes. Thus we need to bump object's Generation (if set). This handles generation bump during graceful deletion. The bump for objects that don't support graceful deletion is handled in pkg/registry/generic/registry/store.go."